impl/src/planning/subplanners.rs

```diff
@@ -232,70 +231,19 @@ impl<'planner> WorkspaceSubplanner<'planner> {
 
   /** Produces a crate context for each declared crate and dependency. */
   fn produce_crate_contexts(&self) -> Result<Vec<CrateContext>> {
-    // Build a list of all catalogs we want the context for
-    let mut catalogs = vec![self.crate_catalog];
-    for catalog in self.binary_dependencies.iter() {
-      catalogs.push(catalog);
-    }
-
-    // Build a list of all workspace files
-    let mut files: Vec<&CargoWorkspaceFiles> = self
-      .binary_deps_files
-      .iter()
-      .map(|(_key, val)| val)
-      .collect();
-    files.push(self.files);
-
-    // Gather the checksums for all packages in the lockfile
-    // which have them.
-    //
-    // Store the representation of the package as a tuple
-    // of (name, version) -> checksum.
-    let mut package_to_checksum = HashMap::new();
-    for workspace_files in files.iter() {
-      if let Some(lock_path) = workspace_files.lock_path_opt.as_ref() {
-        let lockfile = Lockfile::load(lock_path.as_path())?;
-        for package in lockfile.packages {
-          if let Some(checksum) = package.checksum {
-            package_to_checksum.insert(
-              (package.name.to_string(), package.version),
-              checksum.to_string(),
-            );
-          }
-        }
-      }
-    }
-
-    // Additionally, the binary dependencies need to have their checksums added as well in
-    // Remote GenMode configurations. Vendored GenMode relies on the behavior of `cargo vendor`
-    // and doesn't perform any special logic to fetch binary dependency crates.
-    if self.settings.genmode == GenMode::Remote {
-      for (pkg, info) in self.settings.binary_deps.iter() {
-        let version = semver::Version::parse(info.req())?;
-        package_to_checksum.insert(
-          (pkg.clone(), version.clone()),
-          fetch_crate_checksum(&self.settings.index_url, pkg, &version.to_string())?,
-        );
-      }
-    }
-
-    let contexts = catalogs
-      .iter()
-      .map(|catalog| {
-        (*catalog)
-          .metadata
-          .resolve
-          .as_ref()
-          .ok_or_else(|| RazeError::Generic("Missing resolve graph".into()))?
-          .nodes
-          .iter()
-          .sorted_by_key(|n| &n.id)
-          .filter_map(|node| self.create_crate_context(node, &package_to_checksum, catalog))
-          .collect::<Result<Vec<CrateContext>>>()
-      })
-      .collect::<Result<Vec<Vec<CrateContext>>>>()?;
-
-    Ok(contexts.iter().flat_map(|i| i.clone()).collect())
+    Ok(
+      self
+        .crate_catalog
+        .metadata
+        .resolve
+        .as_ref()
+        .ok_or_else(|| RazeError::Generic("Missing resolve graph".into()))?
+        .nodes
+        .iter()
+        .sorted_by_key(|n| &n.id)
+        .filter_map(|node| self.create_crate_context(node, &self.crate_catalog))
+        .collect::<Result<Vec<CrateContext>>>()?,
```

Comment:
I'd prefer more imperative framing of this. I don't think the fluent writing of this is is very readable.

Reply:
This is only a slight modification to what was there before. Can this be deferred or can you write a suggestion?

Reply:
Minor: `Ok(` wrap on something that returns `Result` ? Why not unfold one level?

Reply:
Done